### ISSUE TITLE
`Wrong type argument: stringp, nil` when enabling crafted-compile

### ISSUE THREAD

**Issue by mattmc3** [SystemCrafters/crafted-emacs#235]
I'm running [emacs-plus@28 --with-native-comp](https://github.com/d12frosted/homebrew-emacs-plus) on the latest macOS. When I create an empty `~/.config/crafted-emacs` directory and copy [the example config.el](https://github.com/SystemCrafters/crafted-emacs/blob/master/examples/example-config.el) into it, emacs reports `Wrong type argument: stringp, nil`. If I comment out `(require 'crafted-compile)`, the error goes away, but of course I want this module enabled.

If I `(setq debug-on-error t)`, I get the following:

```lisp
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  file-newer-than-file-p(nil "/Users/matt/.config/crafted-emacs/early-config.el")
  (if (file-newer-than-file-p (crafted-compile-locate-eln-file (file-name-base source)) source) (native-compile-async f) (message "Skipping compilation of file %s" source))
  (let ((source (car --dolist-tail--))) (if (file-newer-than-file-p (crafted-compile-locate-eln-file (file-name-base source)) source) (native-compile-async f) (message "Skipping compilation of file %s" source)) (setq --dolist-tail-- (cdr --dolist-tail--)))
  (while --dolist-tail-- (let ((source (car --dolist-tail--))) (if (file-newer-than-file-p (crafted-compile-locate-eln-file (file-name-base source)) source) (native-compile-async f) (message "Skipping compilation of file %s" source)) (setq --dolist-tail-- (cdr --dolist-tail--))))
  (let ((--dolist-tail-- f)) (while --dolist-tail-- (let ((source (car --dolist-tail--))) (if (file-newer-than-file-p (crafted-compile-locate-eln-file (file-name-base source)) source) (native-compile-async f) (message "Skipping compilation of file %s" source)) (setq --dolist-tail-- (cdr --dolist-tail--)))))
  (if (featurep 'native-compile) (let ((--dolist-tail-- f)) (while --dolist-tail-- (let ((source (car --dolist-tail--))) (if (file-newer-than-file-p (crafted-compile-locate-eln-file (file-name-base source)) source) (native-compile-async f) (message "Skipping compilation of file %s" source)) (setq --dolist-tail-- (cdr --dolist-tail--))))) (let ((--dolist-tail-- f)) (while --dolist-tail-- (let ((source (car --dolist-tail--))) (if (file-exists-p source) (progn (if (file-directory-p source) (byte-recompile-directory source 0) (byte-recompile-file source nil 0)))) (setq --dolist-tail-- (cdr --dolist-tail--))))))
  crafted-compile-file(("/Users/matt/.config/crafted-emacs/early-config.el" "/Users/matt/.config/crafted-emacs/config.el"))
  crafted-compile-config()
  (progn (crafted-compile-config))
  (if crafted-compile-user-configuration (progn (crafted-compile-config)))
  crafted-compile-modules()
  (progn (crafted-compile-modules))
  (if (or crafted-compile-modules crafted-compile-user-modules crafted-compile-init-files crafted-compile-user-configuration) (progn (crafted-compile-modules)))
  (closure (t) nil (if (or crafted-compile-modules crafted-compile-user-modules crafted-compile-init-files crafted-compile-user-configuration) (progn (crafted-compile-modules))))()
  run-hooks(emacs-startup-hook term-setup-hook)
  #f(compiled-function () #<bytecode 0xa4b8f4c971eb100>)()
  normal-top-level()
```

I've tried both with and without an early-config.el file in place. I'm at the limit of my lisp knowledge at this point to try much else.

**Comment by jeffbowman** [comment]
So, it looks like `crafted-compile-locate-eln-file` is returning `nil` for a compiled file name, which causes the error you see when calling `file-newer-than-file-p`.  I don't use the native compile stuff in Emacs 28, but maybe @abarocio80  can help here?

**Comment by abarocio80** [comment]
As far as I understand, It shouldn't return `nil` if the **compiled file** exist for the current version. What does `(expand-file-name comp-native-version-dir native-comp-eln-load-path)` return? It should be a directory. What is the content of that directory? Does the **compiled** `.eln` is there?

Another possibility is that it is a Mac OS or emacs-plus implementation issue.

**Comment by thibautbenjamin** [comment]
I have the same problem on linux and emacs 29. I submitted a PR to handle the case where the function returns `nil` (I actually submitted the PR before reading this issue, so it might not be a good fix at all, but it at least lets me run emacs without error)

**Comment by jeffbowman** [comment]
Closing since I merged #238
